### PR TITLE
[DOXIA-588] The Sink.JUSTIFY_XXX was ignored

### PR DESCRIPTION
**Pull request by qualitesys**
See Jira ticket https://issues.apache.org/jira/browse/DOXIA-588
Source code patch
Update Unit test as well

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+ # Licensed to the Apache Software Foundation (ASF) under one
```
Comment (elharo) [inline .github/workflows/maven.yml:1]: This should be in a separate PR as it's not directly related to fixing this bug

File: doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSink.java [hunk 2]
```diff
@@ -1658,6 +1659,35 @@ public void tableHeaderCell( SinkEventAttributes attributes )
     private void tableCell( boolean headerRow, MutableAttributeSet attributes )
     {
         Tag t = ( headerRow ? HtmlMarkup.TH : HtmlMarkup.TD );
+        
+        if (   !headerRow 
+            && null != this.cellCountStack
```
Comment (elharo) [inline doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSink.java:1664]: this.foo != null
is far more common and unsurprising. Putting the literal first is a C convention not needed in Java.

**Review by elharo — COMMENTED** [review]
running through Jenkins: https://ci-builds.apache.org/job/Maven/job/maven-box/job/maven-doxia/job/DOXIA-588/

**Review by elharo — CHANGES_REQUESTED** [review]
Jenkins passed but .github/workflows/maven.yml needs to be moved to a separate PR

File: .github/workflows/maven.yml (hunk 1 above)
Comment (qualitesys) [reply .github/workflows/maven.yml:1]: I am not a GitHub expert at all. I can remove maven.yml in my repo, no problem. Is it enough for this PR ?

File: doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSink.java (hunk 2 above)
Comment (qualitesys) [reply doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSink.java:1664]: Starting with constants in logical expressions prevents from a very common mistakes that can be very tricky, this concerns Java, C, C++, C# att least.

See [https://cwe.mitre.org/data/definitions/481.html](url)

Comment (elharo) [reply doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSink.java:1664]: In C, yes, Not in Java because == is used for comparison instead of =. The only way this can occur in Java is if you're comparing to a boolean and in that case you should just use the boolean directly.

File: .github/workflows/maven.yml (hunk 1 above)
Comment (elharo) [reply .github/workflows/maven.yml:1]: you need to push too.

**Comment by slachiewicz** [comment]
thx for the patch, I merged it with minor changes. 
I also added changes to Github Actions based on our other projects 794c39dbee17a35e2e74bf001450b2c3cf15959a
